### PR TITLE
NETSCRIPT: Added ns.corporation.issueNewShares function

### PR DESCRIPTION
**Pull request by Mughur**
Added API call for issuing new shares, takes in share amount as an optional parameter, defaults to max amount issuable. Returns the amount of funds generated from the action.

Uses the same (mostly copypasted) logic as the UI action.

### CODE REVIEW

File: src/Corporation/Actions.ts [hunk 1]
```diff
@@ -90,6 +91,27 @@ export function IssueDividends(corporation: Corporation, rate: number): void {
   corporation.dividendRate = rate;
 }
 
+export function IssueNewShares(corporation: Corporation, amount: number, max: number): number {
```
Comment (Snarling) [inline src/Corporation/Actions.ts]: If this is in Corporation/Actions.ts, shouldn't the actual corporation UI use this function? It looks like this is only being used by the new netscript function. Is there a reason both are not using the same function?

File: src/Corporation/Actions.ts [hunk 2]
```diff
@@ -90,6 +91,27 @@ export function IssueDividends(corporation: Corporation, rate: number): void {
...
+  if (isNaN(amount) || amount < 0 || amount > max) {
```
Comment (Snarling) [inline src/Corporation/Actions.ts]: Doesn't issuing new shares also have to be in increments of 10m? That doesn't seem to be enforced here.

**Review by Snarling — CHANGES_REQUESTED** [review]

File: src/Corporation/Actions.ts (hunk 2 above)
Comment (Mughur) [reply src/Corporation/Actions.ts]: aw shucks. I calculated the 10m increment amount for the max new shares only, you're absolutely right

File: src/Corporation/Actions.ts (hunk 1 above)
Comment (Mughur) [reply src/Corporation/Actions.ts]: They do now

**Comment by Snarling** [comment]
I'll merge this after the main gameplay affecting corp PR (you were doing a balance changes PR right?) and the corp consts ones. I foresee possible merge conflicts between the PRs and it should be easier to resolve them on this one since it's smaller.

**Review by Snarling — APPROVED** [review]

**Pull request merged**